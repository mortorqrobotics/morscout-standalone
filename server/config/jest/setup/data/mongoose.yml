Season:
  - _id: 5d7be5a7994e35048b9e5e0e
    year: 2019
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
  - _id: 4d7be5a7994e35048b9e5e0e
    year: 2015
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
Regional:
  - _id: 6d7be5a7994e35048b9e5e0e
    name: Hi!!!
    key: abc
    url: google.com
    eventType: regional
    startDate: 2019-01-08 12:00:00 -5
    endDate: 2019-02-03 12:00:00 -5
    season: 5d7be5a7994e35048b9e5e0e
    location:
      type: Point
      coordinates: [0,0]
Alliance:
  - _id: 2d7ce5a7994e35048b9e5e0e
    teams:
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
Match:
  - _id: 7d7be5a7994e35048b9e5e0e
    redAlliance: 2d7ce5a7994e35048b9e5e0e
    blueAlliance: 2d7ce5a7994e35048b9e5e0e
    winner: Red
    startDate: 2019-02-03 12:00:00 -5
    endDate: 2019-02-03 12:10:00 -5
    Regional: 6d7be5a7994e35048b9e5e0e
User:
  - _id: 8d7be5a7994e35048b9e5e0f
    team: 9d7be5a7994e35048b9e5e0e
    phone: 1334567890
    email: bo@bob.bob
    firstname: Bob
    lastname: Boby
    password: Secret
    username: bob2
Team:
  - _id: 9d7be5a7994e35048b9e5e0e
    regional: 6d7be5a7994e35048b9e5e0e
    number: 1515
    name: MorTorq
    location:
      type: Point
      coordinates: [0,0]
