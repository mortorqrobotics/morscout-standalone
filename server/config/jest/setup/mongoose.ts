import mongoose from "mongoose";
import { MongoMemoryServer } from "mongodb-memory-server";
import getData from "./getData";

let mongoServer: MongoMemoryServer;
export async function setup() {
  mongoServer = new MongoMemoryServer();
  const mongoUri = await mongoServer.getConnectionString();
  (process as any).mongoUri = mongoUri;
  await mongoose.connect(
    mongoUri,
    { useNewUrlParser: true, useCreateIndex: true },
    err => {
      if (err) console.error(err);
    }
  );
  await getData();
  return mongoUri;
}

export async function teardown() {
  mongoServer.stop();
}
