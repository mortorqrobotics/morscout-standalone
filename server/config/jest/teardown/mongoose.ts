import mongoose from "mongoose";

module.exports = () => {
  mongoose.disconnect();
  (mongoose as any).server.stop();
};
