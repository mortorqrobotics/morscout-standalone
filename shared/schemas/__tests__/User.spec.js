import "mockingoose";
import mongoose from "mongoose";
import schema from "../User";

test("User is same throughout", async () => {
  expect.assertions(1);
  expect(schema).toMatchSnapshot();
});

test("User is a schema", async () => {
  expect.assertions(1);
  expect(schema).toBeInstanceOf(mongoose.Schema);
});

test("Compare Password functions", async () => {
  expect.assertions(3);
  const User = mongoose.model("User", schema);
  const user = new User({
    username: "mortorq",
    password: "Bananas",
    firstname: "mor",
    lastname: "torq",
    email: "a@a.com", // really need to stop (emails can be other than .com)
    phone: "1234567890",
    team: new mongoose.Types.ObjectId(),
    position: "alumnus"
  });
  await user.save();
  await expect(user.comparePassword("Bananas")).resolves.toBe(true);
  await expect(user.comparePassword("Barnanas")).resolves.toBe(false);
  await expect(user.comparePassword()).rejects.toThrow();
});
