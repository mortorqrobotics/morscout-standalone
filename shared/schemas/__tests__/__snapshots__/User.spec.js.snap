// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User is same throughout 1`] = `
Schema {
  "$id": 1,
  "_indexes": Array [],
  "_userProvidedOptions": Object {},
  "aliases": Object {},
  "callQueue": Array [],
  "childSchemas": Array [],
  "inherits": Object {},
  "methodOptions": Object {},
  "methods": Object {
    "comparePassword": [Function],
  },
  "nested": Object {},
  "obj": Object {
    "email": Object {
      "required": true,
      "type": [Function],
      "unique": true,
    },
    "firstname": Object {
      "required": true,
      "type": [Function],
    },
    "lastname": Object {
      "required": true,
      "type": [Function],
    },
    "mobileDeviceTokens": Object {
      "default": Array [],
      "type": Array [
        Object {
          "required": true,
          "type": [Function],
        },
      ],
    },
    "parentEmail": [Function],
    "password": Object {
      "required": true,
      "select": false,
      "type": [Function],
    },
    "phone": Object {
      "required": true,
      "type": [Function],
      "unique": true,
    },
    "position": Object {
      "enum": Array [
        "member",
        "leader",
        "mentor",
        "alumnus",
      ],
      "type": [Function],
    },
    "profpicpath": [Function],
    "scoutCaptain": Object {
      "default": false,
      "type": [Function],
    },
    "team": Object {
      "ref": "Team",
      "type": [Function],
    },
    "username": Object {
      "required": true,
      "type": [Function],
      "unique": true,
    },
  },
  "options": Object {
    "_id": true,
    "autoIndex": null,
    "bufferCommands": true,
    "capped": false,
    "discriminatorKey": "__t",
    "id": true,
    "minimize": true,
    "noId": false,
    "noVirtualId": false,
    "read": null,
    "shardKey": null,
    "strict": true,
    "typeKey": "type",
    "validateBeforeSave": true,
    "versionKey": "__v",
  },
  "paths": Object {
    "_id": ObjectId {
      "_index": null,
      "defaultValue": [Function],
      "getters": Array [],
      "instance": "ObjectID",
      "options": Object {
        "auto": true,
        "type": [Function],
      },
      "path": "_id",
      "setters": Array [
        [Function],
      ],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "email": SchemaString {
      "_index": Object {
        "unique": true,
      },
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "isRequired": true,
      "options": Object {
        "required": true,
        "type": [Function],
        "unique": true,
      },
      "originalRequiredValue": true,
      "path": "email",
      "regExp": null,
      "requiredValidator": [Function],
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "firstname": SchemaString {
      "_index": null,
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "isRequired": true,
      "options": Object {
        "required": true,
        "type": [Function],
      },
      "originalRequiredValue": true,
      "path": "firstname",
      "regExp": null,
      "requiredValidator": [Function],
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "lastname": SchemaString {
      "_index": null,
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "isRequired": true,
      "options": Object {
        "required": true,
        "type": [Function],
      },
      "originalRequiredValue": true,
      "path": "lastname",
      "regExp": null,
      "requiredValidator": [Function],
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "mobileDeviceTokens": SchemaArray {
      "$isMongooseArray": true,
      "_index": null,
      "caster": SchemaString {
        "_index": null,
        "enumValues": Array [],
        "getters": Array [],
        "instance": "String",
        "isRequired": true,
        "options": Object {
          "required": true,
        },
        "originalRequiredValue": true,
        "path": "mobileDeviceTokens",
        "regExp": null,
        "requiredValidator": [Function],
        "setters": Array [],
        "validators": Array [
          Object {
            "message": "Path \`{PATH}\` is required.",
            "type": "required",
            "validator": [Function],
          },
        ],
        Symbol(mongoose#schemaType): true,
      },
      "casterConstructor": [Function],
      "defaultValue": [Function],
      "getters": Array [],
      "instance": "Array",
      "options": Object {
        "default": Array [],
        "type": Array [
          Object {
            "required": true,
            "type": [Function],
          },
        ],
      },
      "path": "mobileDeviceTokens",
      "schemaOptions": Object {
        "_id": true,
        "autoIndex": null,
        "bufferCommands": true,
        "capped": false,
        "discriminatorKey": "__t",
        "id": true,
        "minimize": true,
        "noId": false,
        "noVirtualId": false,
        "read": null,
        "shardKey": null,
        "strict": true,
        "typeKey": "type",
        "validateBeforeSave": true,
        "versionKey": "__v",
      },
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "parentEmail": SchemaString {
      "_index": null,
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "options": Object {
        "type": [Function],
      },
      "path": "parentEmail",
      "regExp": null,
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "password": SchemaString {
      "_index": null,
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "isRequired": true,
      "options": Object {
        "required": true,
        "select": false,
        "type": [Function],
      },
      "originalRequiredValue": true,
      "path": "password",
      "regExp": null,
      "requiredValidator": [Function],
      "selected": false,
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "phone": SchemaString {
      "_index": Object {
        "unique": true,
      },
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "isRequired": true,
      "options": Object {
        "required": true,
        "type": [Function],
        "unique": true,
      },
      "originalRequiredValue": true,
      "path": "phone",
      "regExp": null,
      "requiredValidator": [Function],
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "position": SchemaString {
      "_index": null,
      "enumValidator": [Function],
      "enumValues": Array [
        "member",
        "leader",
        "mentor",
        "alumnus",
      ],
      "getters": Array [],
      "instance": "String",
      "options": Object {
        "enum": Array [
          "member",
          "leader",
          "mentor",
          "alumnus",
        ],
        "type": [Function],
      },
      "path": "position",
      "regExp": null,
      "setters": Array [],
      "validators": Array [
        Object {
          "enumValues": Array [
            "member",
            "leader",
            "mentor",
            "alumnus",
          ],
          "message": "\`{VALUE}\` is not a valid enum value for path \`{PATH}\`.",
          "type": "enum",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "profpicpath": SchemaString {
      "_index": null,
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "options": Object {
        "type": [Function],
      },
      "path": "profpicpath",
      "regExp": null,
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "scoutCaptain": SchemaBoolean {
      "_index": null,
      "defaultValue": false,
      "getters": Array [],
      "instance": "Boolean",
      "options": Object {
        "default": false,
        "type": [Function],
      },
      "path": "scoutCaptain",
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "team": ObjectId {
      "_index": null,
      "getters": Array [],
      "instance": "ObjectID",
      "options": Object {
        "ref": "Team",
        "type": [Function],
      },
      "path": "team",
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "username": SchemaString {
      "_index": Object {
        "unique": true,
      },
      "enumValues": Array [],
      "getters": Array [],
      "instance": "String",
      "isRequired": true,
      "options": Object {
        "required": true,
        "type": [Function],
        "unique": true,
      },
      "originalRequiredValue": true,
      "path": "username",
      "regExp": null,
      "requiredValidator": [Function],
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
  },
  "plugins": Array [],
  "query": Object {},
  "s": Object {
    "hooks": Kareem {
      "_posts": Map {},
      "_pres": Map {
        "save" => Array [
          Object {
            "fn": [Function],
            "isAsync": false,
          },
          Object {
            "fn": [Function],
            "isAsync": false,
          },
        ],
      },
    },
  },
  "singleNestedPaths": Object {},
  "statics": Object {},
  "subpaths": Object {},
  "tree": Object {
    "_id": Object {
      "auto": true,
      "type": [Function],
    },
    "email": Object {
      "required": true,
      "type": [Function],
      "unique": true,
    },
    "firstname": Object {
      "required": true,
      "type": [Function],
    },
    "lastname": Object {
      "required": true,
      "type": [Function],
    },
    "mobileDeviceTokens": Object {
      "default": Array [],
      "type": Array [
        Object {
          "required": true,
          "type": [Function],
        },
      ],
    },
    "parentEmail": [Function],
    "password": Object {
      "required": true,
      "select": false,
      "type": [Function],
    },
    "phone": Object {
      "required": true,
      "type": [Function],
      "unique": true,
    },
    "position": Object {
      "enum": Array [
        "member",
        "leader",
        "mentor",
        "alumnus",
      ],
      "type": [Function],
    },
    "profpicpath": [Function],
    "scoutCaptain": Object {
      "default": false,
      "type": [Function],
    },
    "team": Object {
      "ref": "Team",
      "type": [Function],
    },
    "username": Object {
      "required": true,
      "type": [Function],
      "unique": true,
    },
  },
  "virtuals": Object {},
}
`;
